### DISCUSSION TITLE
Force backward compatible output format? Server is Mysql 8, but I wan't to be able to use mysql-client 5.7

### DISCUSSION THREAD

**Discussion by jihu** [mydumper/mydumper#1560] [Q&A]
So, our database is MySQL 8, but the mysql-client installed on on of our main machines is version 5.7 (and I haven't been able to update it, because it's on Ubuntu 18).

When I use mydumper on this database, it seems to output in a format that isn't compatible with mysql-client. The error I get is:

`ERROR 1064 (42000) at line 4: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '[config]
quote_character = BACKTICK

[myloader_session_variables]
SQL_MODE='NO_A' at line 1`

Is there a way to force mydumper to output in a backward compatible format, that mysql-client 5.7 can handle?

**Comment by davidducos** [comment]
I think that you are trying to import the metadata file which is not a sql file. Why don't you use myloader to import the sql file in parallel?

**Reply by jihu** [reply]
ok, strange. But it would make sense that I made a mistake in the export since this trivial use case (export to an sql file) doesn't seem to be mentioned in the documentation. :)

I used the command this way:

`mydumper -u someUserName -a -h someHostName someDatabaseName > output.sql`

What should I have written instead?

I wanted to test if this tool can be used to create a backup, and I wanted to verify the resulting file by importing it using the regular mysql way, using the mysql-client.

**Reply by jihu** [reply]
Oh, one more thing... The resulting file is 1.5GB, which is about the same size as our regular dump files. So I don't think it was just some metadata.

**Reply by davidducos** [reply]
Hi @jihu, 
mydumper doesn't export a single sql file, it exports a backup file compatible only with myloader. 
Seems that you are using mydumper as mysqldump, that is not going to work.

What I meant is that in the backup file, there is metadata content from the metadata file.

**Reply by jihu** [reply]
ok. The description described it as "a tool used for backing up MySQL database servers much faster than the mysqldump tool", and it never said anything about it using a different format.

But regardless of that, is there a reason why the most simple/basic use case isn't documented? As in, a simple minumal concrete example showing how to use the mydumper tool to create a backup to a file, and how to use myloader to import it.

**Reply by davidducos** [reply]
I apologize that the documentation doesn't meet your expectations. Feel free to create a PR to make it better for new users from the experience that you are going to get.

**Reply by jihu** [reply]
I'm sorry, but it makes no sense that a user who doesn't know how to use the software should document how to use the software.
